### PR TITLE
add support of 'six.with_metaclass' 

### LINKED ISSUE #713: Incorrect inference of ancestors of Enum class with six.with_metaclass
### Steps to reproduce
```python
import astroid

code = '''import six
from enum import Enum, EnumMeta

class FooMeta(EnumMeta):
    pass

class Foo(six.with_metaclass(FooMeta, Enum)):
    bar = 1'''

module = astroid.parse(code)
print(list(module.body[3].ancestors()))
```

### Current behavior
Prints `[<ClassDef.NoneType l.0 at 0x102bb4d68>, <ClassDef.object l.0 at 0x102bbe1d0>]`

### Expected behavior
Prints `[<ClassDef.Enum l.10 at 0x10d0617b8>, <ClassDef.object l.0 at 0x10c4650f0>]`

With another simple example without enums works ok:

```python
import astroid

code = '''import six

class FooMeta(type):
    pass

class BaseFoo(object):
    pass

class Foo(six.with_metaclass(FooMeta, BaseFoo)):
    bar = 1'''

module = astroid.parse(code)
print(list(module.body[3].ancestors()))
```
Prints: `[<ClassDef.BaseFoo l.6 at 0x1091531d0>, <ClassDef.object l.0 at 0x10853d128>]`

### ``python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"`` output
2.3.2

This behaviour breaks pylint with exception:
```
Traceback (most recent call last):
  File "bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
  File "lib/python3.7/site-packages/pylint/__init__.py", line 19, in run_pylint
    Run(sys.argv[1:])
  File "lib/python3.7/site-packages/pylint/lint.py", line 1394, in __init__
    linter.check(args)
  File "lib/python3.7/site-packages/pylint/lint.py", line 801, in check
    self._do_check(files_or_modules)
  File "lib/python3.7/site-packages/pylint/lint.py", line 938, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "lib/python3.7/site-packages/pylint/lint.py", line 1018, in check_astroid_module
    walker.walk(ast_node)
  File "lib/python3.7/site-packages/pylint/utils.py", line 1162, in walk
    self.walk(child)
  File "lib/python3.7/site-packages/pylint/utils.py", line 1162, in walk
    self.walk(child)
  File "lib/python3.7/site-packages/pylint/utils.py", line 1162, in walk
    self.walk(child)
  File "lib/python3.7/site-packages/pylint/utils.py", line 1159, in walk
    cb(astroid)
  File "lib/python3.7/site-packages/pylint/checkers/base.py", line 1509, in visit_assignname
    if not list(frame.local_attr_ancestors(node.name)):
  File "lib/python3.7/site-packages/astroid/scoped_nodes.py", line 2197, in local_attr_ancestors
    if name in astroid:
TypeError: argument of type 'Const' is not iterable
```

### PR DESCRIPTION
**Pull request by fmigneault**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Fixes issue #713 by providing corresponding transform and unit tests.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |
| ✓  | :sparkles: New feature |

## Related Issue

Closes #713

### CODE REVIEW

**Comment by fmigneault** [comment]
flagged pylint issues are related to very recent merge of PyCQA/pylint#3782 with fix of issue PyCQA/pylint#3768

**Comment by tkukushkin** [comment]
@fmigneault please pay attention to https://github.com/PyCQA/astroid/issues/713#issuecomment-558166958. This code

```python
from enum import Enum, EnumMeta

import six


class FooMeta(EnumMeta):
    pass


class Foo(six.with_metaclass(FooMeta, Enum)):
    bar = 1
```

produces error with this fix:
```
foo.py:3: [W0611 unused-import] Unused import six
```

**Comment by fmigneault** [comment]
@tkukushkin 
I don't know enough the internals of `astroid` to address that specific issue of  https://github.com/PyCQA/astroid/pull/841#issuecomment-721769142. Maybe @PCManticore (or other dev from `astroid`) can help?

The problem in is that after `six.with_metaclass(FooMeta, Enum)` is processed, the `ClassDef` effectively becomes: 
```
ClassDef.FooMeta(name='FooMeta',
                 doc=None,
                 decorators=None,
                 bases=[<Name.EnumMeta l.6 at 0x7ff0d24e1a10>],
                 body=[<Pass l.7 at 0x7ff0d24e1a90>])
```
Therefore there is no more reference to `six` as seen in the source code. At runtime, the class definition is "converted" and doesn't need `six` anymore. Is there some way to maintain that reference somewhere in the `ClassDef` for the check to find it?

**Comment by tkukushkin** [comment]
@fmigneault I completely understand the cause of this problem and I don't know not hacky solutions and PCManticore did not respond to my comment. That's why I didn't make pull request with this transform.

File: astroid/brain/brain_six.py [hunk 1]
```diff
@@ -189,6 +189,29 @@ def transform_six_add_metaclass(node):
...
 
 
+def _looks_like_nested_from_six_with_metaclass(node):
+    if len(node.bases) != 1:
+        return False
+    base = node.bases[0]
+    if not isinstance(base, nodes.Call):
+        return False
+    try:
+        func = next(base.func.infer())
```
Comment (PCManticore) [inline astroid/brain/brain_six.py:199]: Unfortunately we cannot use inference in predicate functions. Here you would need o unpack the arguments for checking if they refer to `six.with_metaclass` or not (by unpacking I mean either checking explicitly for `Name` or `Attribute` instances that match against `six.with_metaclass`)

**Review by PCManticore — CHANGES_REQUESTED** [review]
Overall the change looks good, but we need to drop the inference check from the predicate function as it can result in weird inference bugs.

File: astroid/brain/brain_six.py (hunk 1 above)
Comment (fmigneault) [reply astroid/brain/brain_six.py]: Should be good now. Let me know if it is still not what like you meant.

**Review by PCManticore — APPROVED** [review]

**Comment by PCManticore** [comment]
Thanks for the PR @fmigneault !

**Pull request merged**